File: src/lessons/welcome/bat/bool1/Close10Entity.java
package lessons.welcome.bat.bool1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class Close10Entity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(close10((Integer)param[0], (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  int close10(int a, int b)
  {
    /* BEGIN SOLUTION */
    if (Math.abs(a - 10) == Math.abs(b - 10))
      return 0;
    if (Math.abs(a - 10) < Math.abs(b - 10))
      return a;
    return b;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool1/Close10Entity.scala
package lessons.welcome.bat.bool1

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class Close10Entity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(close10(param(0).asInstanceOf[Int],param(1).asInstanceOf[Int])))
	  }
	}

	/* BEGIN TEMPLATE */
	def close10(a:Int, b:Int): Int = {
	/* BEGIN SOLUTION */
   if (Math.abs(10-a) == Math.abs(10-b))
      return 0
   else if (Math.abs(10-a) < Math.abs(10-b))
      return a
   else
      return b
	/* END SOLUTION */
	}
	/* END TEMPLATE */
}
